altair/vegalite/v5/api.py

```diff
@@ -1068,12 +1078,22 @@ def when(
             )
             raise NotImplementedError(msg)
 
-    def to_dict(self, *args: Any, **kwds: Any) -> _Conditional[_C]:  # type: ignore[override]
-        m = super().to_dict(*args, **kwds)
-        return _Conditional(condition=m["condition"])
+    def to_dict(self, *args: Any, **kwds: Any) -> _Conditional[_C]:
+        return _Conditional(condition=self.condition.copy())
 
     def __deepcopy__(self, memo: Any) -> Self:
-        return type(self)(_Conditional(condition=_deepcopy(self.condition)))
+        return type(self)(_Conditional(condition=_deepcopy(self.condition, memo)))
+
+    def __repr__(self) -> str:
+        name = type(self).__name__
+        COND = "condition: "
+        LB, RB = "{", "}"
+        if len(self.condition) == 1:
+            args = f"{COND}{self.condition!r}".replace("\n", "\n  ")
+        else:
+            conds = "\n    ".join(f"{c!r}" for c in self.condition)
+            args = f"{COND}[\n    " f"{conds}\n  ]"
+        return f"{name}({LB}\n  {args}\n{RB})"
```

Comment:
For reference, this is the short/long form 

![image](https://github.com/user-attachments/assets/7268c6d0-b6eb-48e8-bf6d-46da789c2c3b)

Reply:
Nice! :)